### ISSUE TITLE
More ergonomic testing API

### ISSUE THREAD

**Issue by NoraCodes** [xi-editor/xi-editor#1169]
Per [this proposal](https://xi-editor.io/xi-editor/gsoc.html#flush-out-unit-tests), I'd like to look at designing and implementing a more ergonomic scripting API, based on a more in-depth version of [fluent ergonomic testing APIs](https://nora.codes/tutorial/building-readable-tests-with-fluent-testing-apis/).

### The Problem

It is difficult to write (and read!) very in-depth tests for Xi's core functions because there is no existing testing API. Because of this there are not all the tests we would like to have.

### The Solution

Rather than writing a DSL, as suggested, I would like to implement a very readable and concise API, which is just regular Rust. This avoids any kind of parsing issues, and will make implementation of tests using existing Rust tooling (`cargo fmt`, RLS, etc) so no custom tooling will be needed.

**Comment by cmyr** [comment]
I agree we should have an issue for this, it's definitely an important project.
